control-plane-operator/controllers/hostedcontrolplane/v2/kas/config.go

```diff
@@ -354,6 +353,22 @@ func admissionPlugins() []string {
 		"security.openshift.io/ValidateSecurityContextConstraints",
 		"storage.openshift.io/CSIInlineVolumeSecurity",
 	}
+
+	if util.ConfigOAuthEnabled(cfg.Authentication) {
+		enabled = append(enabled, "authorization.openshift.io/RestrictSubjectBindings", "authorization.openshift.io/ValidateRoleBindingRestriction")
+	}
+
+	return enabled
+}
+
+func disabledAdmissionPlugins(cfg KubeAPIServerConfigParams) []string {
+	disabled := []string{}
+
+	if !util.ConfigOAuthEnabled(cfg.Authentication) {
```

Comment:
The automated fixtures tests are not covering config generation for this path.
Let's make sure we cover it. We could add coverage with the fixtures (see TestControlPlaneComponents) but that would result in many duplicated fixtures.
I think we can probably just add a unit for generateConfig()

cc @muraee in case you had other plans for adding new test cases such as this one.

Reply:
Also is there any e2e in place validating this change?

Reply:
yeah, adding coverage in TestControlPlaneComponents will introduce too many duplicates . I think this should be covered in new local unit test for generateConfig() indeed.

Reply:
>Also is there any e2e in place validating this change?

No e2e tests in place. IIUC the only case where the OpenShift OAuth stack would be disabled is when the external OIDC configuration is specified and there is no pre-existing e2e tests for the external OIDC behavior.

I have plans to come back around to implementing e2e tests for the external OIDC feature, but if they _must_ be implemented before this can merge then I'll jump to that and return to this PR later.

I can add a new unit test for this change.